### PR TITLE
Insights: formalize insight series types

### PR DESCRIPTION
**Pull request by coury-clark**
Stacked on https://github.com/sourcegraph/sourcegraph/pull/28492

Closes https://github.com/sourcegraph/sourcegraph/issues/27435

Adds a new column to track how an insight series is generated. This will replace some ad-hoc logic around our app with something a bit more formalized.

### CODE REVIEW

File: migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+
+ALTER TABLE insight_series
+    DROP COLUMN generation_method;
+ALTER TABLE insight_series
+    DROP COLUMN just_in_time;
```
Comment (CristinaBirkel) [inline migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql:6]: Should we add `IF EXISTS` to these?

**Review by CristinaBirkel — APPROVED** [review]
Looks good! 👍

File: migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql (hunk 1 above)
Comment (coury-clark) [reply migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql:6]: yeah, good idea!

File: migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+
+ALTER TABLE insight_series
+    DROP COLUMN generation_method;
```
Comment (coury-clark) [inline migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql:4]:
```suggestion
    DROP COLUMN IF EXISTS generation_method;
```

File: migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql (hunk 1 above)
Comment (coury-clark) [inline migrations/codeinsights/1000000024_insight_series_expanded_type_info.down.sql:6]:
```suggestion
    DROP COLUMN IF EXISTS just_in_time;
```

**Pull request merged**